dataset/dataset.cpp

```diff
@@ -216,6 +216,22 @@ void Dataset::setData(const std::string &name, Variable data,
   setData_impl(name, detail::DatasetData{std::move(data), {}, {}}, attrPolicy);
 }
 
+/// Return the size in memory of the Dataset.
+/// To avoid double counting and the unecessary
+/// creation of view objects works with the
+/// private data containers.
+scipp::index Dataset::sizeInMemory() const {
```

Comment:
Is there any reason this has to be a method? Keeping the class interface smaller may be advantage. That way, you could probably even use the same implementation for views and datasets (or views and data arrays, and views and variables): You can simply compute the sizes based on objects `dims()` and other publicly accessible properties?